clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -329,17 +329,18 @@ private TransactionalRequestResult beginCompletingTransaction(TransactionResult
     }
 
     public synchronized TransactionalRequestResult sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offsets,
```

Comment:
We unify the API for groupId and groupMetadata commit here, which includes the groupId within the metadata struct.
A boolean flag indicating whether to turn on global fencing shall be passed down to the txn commit sender to determine whether we should include `(member.id, instance.id, generation.id)` in the request.